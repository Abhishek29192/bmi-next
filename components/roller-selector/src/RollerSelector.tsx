import React from "react";
import classnames from "classnames";
import ButtonBase, { ButtonBaseProps } from "@material-ui/core/ButtonBase";
import ArrowIcon from "@material-ui/icons/ArrowForward";
import styles from "./RollerSelector.module.scss";

type Props = {
  children: React.ReactNode;
  component?: React.ComponentType<any>; // TODO
  isSelected?: boolean;
} & ButtonBaseProps;

const RollerSelector = ({
  children,
  component: Component = ButtonBase,
  isSelected,
  onClick,
  className,
  ...rest
}: Props) => {
  return (
    <Component
      onClick={onClick}
      className={classnames(
        className,
        styles["RollerSelector"],
        isSelected && styles["RollerSelector--selected"]
      )}
      {...rest}
    >
      {children}
      <ArrowIcon className={styles["icon"]} />
    </Component>
  );
};
export default RollerSelector;
