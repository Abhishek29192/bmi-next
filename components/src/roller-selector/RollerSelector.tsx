import React from "react";
import classnames from "classnames";
import { ButtonBase, ButtonBaseProps } from "@material-ui/core";
import { ArrowForward } from "@material-ui/icons";
import styles from "./RollerSelector.module.scss";

type Props = {
  children: React.ReactNode;
  component?: React.ComponentType<any>; // TODO
  isSelected?: boolean;
} & ButtonBaseProps;

const RollerSelector = ({
  children,
  component: Component = ButtonBase,
  isSelected,
  onClick,
  className,
  ...rest
}: Props) => {
  return (
    <Component
      onClick={onClick}
      className={classnames(
        className,
        styles["RollerSelector"],
        isSelected && styles["RollerSelector--selected"]
      )}
      {...rest}
    >
      {children}
      <ArrowForward className={styles["icon"]} />
    </Component>
  );
};
export default RollerSelector;
