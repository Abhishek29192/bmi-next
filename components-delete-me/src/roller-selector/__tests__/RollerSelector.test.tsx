import React from "react";
import { fireEvent } from "@testing-library/react";
import RollerSelector from "../RollerSelector";
import { renderWithThemeProvider } from "../../__tests__/helper";

describe("RollerSelector component", () => {
  it("renders correctly", () => {
    const { container } = renderWithThemeProvider(
      <RollerSelector>Lorem ipsum sit dolor</RollerSelector>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders selected", () => {
    const { container } = renderWithThemeProvider(
      <RollerSelector isSelected>Lorem ipsum sit dolor</RollerSelector>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders with custom className", () => {
    const { container } = renderWithThemeProvider(
      <RollerSelector className="testing">Lorem ipsum sit dolor</RollerSelector>
    );
    expect(container).toMatchSnapshot();
  });
  it("triggers an onClick event", () => {
    const onClick = jest.fn();
    const label = "Lorem ipsum sit dolor";
    const { getByText } = renderWithThemeProvider(
      <RollerSelector>{label}</RollerSelector>
    );

    fireEvent.click(getByText(label));

    expect(onClick.mock.calls).toMatchSnapshot();
  });
});
