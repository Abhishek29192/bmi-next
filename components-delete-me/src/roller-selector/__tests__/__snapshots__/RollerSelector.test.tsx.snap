// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RollerSelector component renders correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root RollerSelector-root-1"
    tabindex="0"
    type="button"
  >
    Lorem ipsum sit dolor
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root RollerSelector-icon-3"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`RollerSelector component renders selected 1`] = `
<div>
  <button
    class="MuiButtonBase-root RollerSelector-root-4 RollerSelector-selected-5"
    tabindex="0"
    type="button"
  >
    Lorem ipsum sit dolor
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root RollerSelector-icon-6"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`RollerSelector component renders with custom className 1`] = `
<div>
  <button
    class="MuiButtonBase-root testing RollerSelector-root-7"
    tabindex="0"
    type="button"
  >
    Lorem ipsum sit dolor
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root RollerSelector-icon-9"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`RollerSelector component triggers an onClick event 1`] = `Array []`;
